types/session_pool/session_pool.d.ts

```diff
@@ -0,0 +1,213 @@
+/// <reference types="node" />
```

Comment:
the reference to node types here is making TSC confused with "import EventEmitter" below (that is a concrete constructor, but EventEmitter is a namespace when used globally). I'd "whitelist" places that use `EventEmitter` to not include the triple slash at the top

Reply:
Done. Does this resolve the problem with tests above (Causing flaky tests)?